rest-api/dao/participant_summary_dao.py

```diff
@@ -379,6 +381,77 @@ def to_client_json(self, model):
 
     return result
 
+  def make_csv(self, results):
+    csv_data = StringIO.StringIO()
+    writer = unicode_csv.UnicodeWriter(csv_data)
+    headers_list = []
+    write_headers = True
+    for row in results.items:
+      line = self.to_client_csv(row)
+      values_list = []
+
+      if len(headers_list) == 0:
+        for k in line.keys():
+          headers_list.append(k)
+      else:
+        for k in line.keys():
```

Comment:
So my assumption has been that a significant chunk of what's slowing down our current JSON generation is hashing dozens of keys for every result returned from our API.My hope was by returning CSV this wouldn't be necessary anymore -- we'd just append values in a fixed order that matches the headers, and have tests to verify we did it right. Does that work? Have you tested how things perform this way vs. the other way?

But if we were going to do things this way, it seems like we could just call to_client_json and get rid of to_client_csv, and then look things up in the dict when appending to the CSV. Or barring that, reuse format_json_* instead of making format_csv_*....